polars/polars-core/src/datatypes.rs

```diff
@@ -31,52 +24,81 @@ pub struct ListType {}
 pub struct CategoricalType {}
 
 pub trait PolarsDataType: Send + Sync {
+    type ARRAY: Array;
+
     fn get_dtype() -> DataType;
 }
 
 macro_rules! impl_polars_datatype {
-    ($ca:ident, $variant:ident) => {
+    ($ca:ident, $variant:ident, $physical:ty) => {
+        pub struct $ca {}
```

Comment:
This is an important difference in this PR: arrow2 does not have structs declaring types, but `PolarsDataType` is a sentinel type for `ChunkedArray<T>`. Thus, we declare here the empty structs supporting the polars behavior.